client/packages/common/src/utils/regex/RegexUtils.test.ts

```diff
@@ -112,3 +112,27 @@ describe('formatTemplateString', () => {
     );
   });
 });
+
+describe('Check if email is valid', () => {
+  it('rejects empty space in email', () => {
+    expect(RegexUtils.checkEmailIsValid('t est@email.com')).toBeFalsy;
+    expect(RegexUtils.checkEmailIsValid(' test@email.com')).toBeFalsy;
+    expect(RegexUtils.checkEmailIsValid('test@email.com ')).toBeFalsy;
+  });
+  it('rejects email with no @ symbol', () => {
+    expect(RegexUtils.checkEmailIsValid('testemail.com')).toBeFalsy;
+  });
+  it('rejects email with no domain', () => {
+    expect(RegexUtils.checkEmailIsValid('test@email')).toBeFalsy;
+    expect(RegexUtils.checkEmailIsValid('test@email.')).toBeFalsy;
+  });
+  it('rejects empty string before @ symbol', () => {
+    expect(RegexUtils.checkEmailIsValid('@email.com')).toBeFalsy;
+  });
+  it('rejects blank email', () => {
+    expect(RegexUtils.checkEmailIsValid('')).toBeFalsy;
+  });
+  it('accepts correctly formatted email', () => {
+    expect(RegexUtils.checkEmailIsValid('test@email.com')).toBeTruthy;
```

Comment:
Awesome test cases! Just need to call the `toBeFalsy()`/`toBeTruthy()` functions, so that the assertions actually run 😁 

Also might be good to add one for different/funkier emails (e.g. test.user_1@msupply.foundation) - the long domain extension should be okay, but does fail on some email regexes so is a good one to test!

Maybe also things like multiple @@, mulitple ..... in the domain name, etc? Bit of a rabbithole - AI might suggest some good options to validate?

Reply:
Thank you! 😄  Added here b2e6feb422a5fcc33a677d463755e616eba84a59